packages/context-editor/src/plugins/structureDecorations.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { widget } from "@handlewithcare/react-prosemirror";
+import { Plugin } from "prosemirror-state";
+import { Decoration, DecorationSet } from "prosemirror-view";
+
+import { BlockDecoration, InlineDecoration } from "../components/StructureDecoration";
+
+export default () => {
+	return new Plugin({
+		props: {
+			decorations: (state) => {
+				const decorations: Decoration[] = [];
+				state.doc.descendants((node, pos) => {
+					if (node.type.isBlock) {
+						// TODO: is there a better key we can use?
+						decorations.push(widget(pos, BlockDecoration, { key: `node-${pos}` }));
```

Comment:
this file isn't all too different from the original `structureDecorations.ts`. It changes rendering the widget via React thanks the `widget()` from `react-prosemirror`. That is `StructureDecoration.tsx` instead of the JS `wrapWidget` function from before